File: tutorial/05-arrays/problems/04-max-and-second-max.ts
class Soft05p04 {
  solution(arr: number[]) {
    let max = 0;
    let secMax = -1;
    for (let i = 0; i < arr.length; i++) {
      if (max < arr[i]) {
        secMax = max;
        max = arr[i];
      } else {
        if (arr[i] != max) {
          secMax = Math.max(secMax, arr[i]);
        }
      }
    }
    return [max, secMax];
  }
}

File: tutorial/05-arrays/problems/04-max-and-second-max.py
class Soft05p04:
    def solution(self,arr):
        mx,secMax=0,-1
        for i in range(len(arr)):
            if mx <  arr[i]:
                secMax= mx
                mx=arr[i]
            else:
                if mx != arr[i]:
                    secMax= max(secMax,arr[i])
        return [mx,secMax]
arr=[4,4]
soft= Soft05p04()
print(soft.solution(arr))